### PR TITLE
APEXCORE-451 : get-app-package-operators in ApexCLI to contain "type"…

### PR DESCRIPTION
**Pull request by shubham-pathak22**
… property to indicate operator or module

### CODE REVIEW

**Comment by shubham-pathak22** [comment]
Hi, @tushargosavi @PramodSSImmaneni @tweise @davidyan74  Could you please review the changes.

File: engine/src/main/java/com/datatorrent/stram/webapp/TypeGraph.java [hunk 1]
```diff
@@ -348,13 +348,13 @@ public int size()
...
   {
-    TypeGraphVertex tgv = typeGraph.get(Operator.class.getName());
+    TypeGraphVertex tgv = typeGraph.get(DAG.GenericOperator.class.getName());
     if (tgv == null) {
       return null;
     }
     Set<String> result = new TreeSet<>();
     for (TypeGraphVertex node : tgv.allInstantiableDescendants) {
-      if ((isAncestor(InputOperator.class.getName(), node.typeName) || !getAllInputPorts(node).isEmpty())) {
+      if ((isAncestor(DAG.GenericOperator.class.getName(), node.typeName) || !getAllInputPorts(node).isEmpty())) {
```
Comment (davidyan74) [inline engine/src/main/java/com/datatorrent/stram/webapp/TypeGraph.java]: Can you explain why this change? It looks like it's changing the intention of the original code.

File: engine/src/main/java/com/datatorrent/stram/webapp/OperatorDiscoverer.java [hunk 2]
```diff
@@ -526,6 +544,17 @@ public JSONObject describeOperator(String clazz) throws Exception
         response.put(PORT_TYPE_INFO_KEY, portTypeInfo);
         response.put("inputPorts", inputPorts);
         response.put("outputPorts", outputPorts);
+        String type = null;
+        Class<?> genericOperator = classLoader.loadClass(clazz);
+        if (Module.class.isAssignableFrom(genericOperator)) {
+          type = GenericOperatorType.MODULE.getType();
+        }
+        if (Operator.class.isAssignableFrom(genericOperator)) {
```
Comment (davidyan74) [inline engine/src/main/java/com/datatorrent/stram/webapp/OperatorDiscoverer.java]: else if?

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java [hunk 3]
```diff
@@ -2963,7 +2964,7 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
       String[] newArgs = new String[args.length - 1];
       System.arraycopy(args, 1, newArgs, 0, args.length - 1);
       GetOperatorClassesCommandLineInfo commandLineInfo = getGetOperatorClassesCommandLineInfo(newArgs);
-      String parentName = commandLineInfo.parent != null ? commandLineInfo.parent : Operator.class.getName();
+      String parentName = commandLineInfo.parent != null ? commandLineInfo.parent : DAG.GenericOperator.class.getName();
       String files = expandCommaSeparatedFiles(commandLineInfo.args[0]);
```
Comment (tushargosavi) [inline engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java:2968]: you can import DAG.GenericOperator, and use GenericOperator directly.

File: engine/src/main/java/com/datatorrent/stram/webapp/TypeGraph.java (hunk 1 above)
Comment (tweise) [inline engine/src/main/java/com/datatorrent/stram/webapp/TypeGraph.java]: That's right, it was either input operator or at least one input port.

Comment (shubham-pathak22) [inline engine/src/main/java/com/datatorrent/stram/webapp/TypeGraph.java]: You are right. This is not needed. I assumed getAllDTInstantiableOperators() was where all operators are loaded into the result set . Will revert it.

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java (hunk 3 above)
Comment (shubham-pathak22) [inline engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java:2968]: Alright. Will do so

**Comment by shubham-pathak22** [comment]
Incorporated review comments.

**Comment by pramodin** [comment]
Looks good to me, @davidyan74 can you also check

**Comment by davidyan74** [comment]
looks fine
